### DISCUSSION TITLE
Cannot round symbolic expression

### DISCUSSION THREAD

**Discussion by hongguoadp** [MilesCranmer/PySR#583] [Q&A]
Hi
Please help on this exception, i have it defined as :

```python
unary_operators=[
    #"neg",
    "abs",
    "rd(x) = round(x+Float32(0.00001), digits=2)",
    #"floor", 'ceil', 
    'sign'
    # ^ Custom operator (julia syntax)
],
extra_sympy_mappings={
    'rd': lambda x: round(x+0.00001, 2)
}
```

But after long run, it failed with 
"Cannot round symbolic expression"

Thanks
Andrew

**Comment by MilesCranmer** [comment]
Your `extra_sympy_mappings` version of `rd` isn't a valid sympy function. To fix this:

```python
extra_sympy_mappings={
    'rd': lambda x: sympy.ceiling((x+0.00001)*100 - 0.5) / 100
}
```
I think this should fix it. But you should test it first.

Edited: Fixed to `ceiling`.

**Comment by hongguoadp** [comment]
Thanks, but I get this now:

return sympy.ceil((x+0.00001)*100 - 0.5) [/](https://file+.vscode-resource.vscode-cdn.net/) 100

AttributeError: module 'sympy' has no attribute 'ceil'

I did 
import sympy

**Reply by MilesCranmer** [reply]
Oops sorry it's `sympy.ceiling`.

```python
In [13]: import sympy

In [14]: from sympy.abc import x, y, z

In [15]: sympy.ceiling((x + 0.0001)*100 - 0.5) / 100
Out[15]: ceiling(100*x - 0.49)/100
```

**Reply by MilesCranmer** [reply]
Basically sympy doesn't have a round. It only has ceiling. So you can't use `round(...)` when creating your sympy mapping; you have to use `sympy.ceiling`.
